katas/args/jhon/parser.js

```diff
@@ -0,0 +1,101 @@
+export class Parser {
+    constructor() {
+        this.args = process.argv[2];
+        this.schema = {
+            param0 :{
+                id: 0,
+                symbol: 'l',
+                type: 'boolean'
+            },
+            param1 :{
+                id: 1,
+                symbol: 'p',
+                type: 'int'
+            },
+            param2 :{
+                id: '2',
+                symbol: 'd',
+                type: 'String'
+            }
+        };
+    }
+
+    parser(args) {
+        let res = {
+            l : false,
+            p : 0,
+            d : ''
+        };
+        let pos = 0;
+        while (pos < args.length) {
+            let char = args.charAt(pos);
+            if (char == '-') {
+                char = args.charAt(pos = pos + 1);
+                let charR = args.charAt(pos = pos + 1);
+                if (this.isCommand(char) && charR == ' ' && args.charAt(pos + 1) != '-') {
+                    let cont = this.getValue(args, pos);
+                    pos = pos + cont.length + 2;
+                    res[char] = this.fillSchema(cont, char);
+                    console.log(res);
+                } else {
+                    res[char] = this.fillSchema('', char);
+                    pos++;
+                }
+            } else {
+                console.log('Inserte argumentos validos');
+                break;
+            }
+        }
+        return res;
+    }
+
+    isCommand(char) {
+        let flag = Boolean(false);
+        for (let x = 0; x < 3; x++) {
+            let args = 'param' + x;
+            let charAux = this.schema[args]['symbol'];
+            if (char == charAux) {
+                flag = Boolean(true);
+            }
+        }
+        return flag;
+    }
+
+    getValue(args, pos) {
+        let res = '';
+        for (let x = pos + 1; x < args.length; x++) {
+            if (args.charAt(x) != ' ') {
+                res = res + args.charAt(x);
+            } else {
+                x = args.length;
+            }
+        }
+        return res;
+    }
+
+    fillSchema(cont, command) {
+        let flag;
+        if (command == 'l') {
+            if (cont.length <= 0) {
+                flag = Boolean(true);
+            } else {
+                flag = Boolean(false);
```

Comment:
You can simplify your code by using ternaries
`flag = cont.length <= 0 ? true : false`
furthermore, in this particular case, since flag is expecting a boolean value, you don't even need to send a boolean, condition already returns one!
`flag = cont.length <= 0`